tokio/src/sync/notify.rs

```diff
@@ -269,30 +284,48 @@ struct AtomicNotification(AtomicUsize);
 
 impl AtomicNotification {
     fn none() -> Self {
-        AtomicNotification(AtomicUsize::new(NOTIFICATION_NONE))
+        AtomicNotification(AtomicUsize::new(0))
     }
 
     /// Store-release a notification.
     /// This method should be called exactly once.
-    fn store_release(&self, notification: Notification) {
-        self.0.store(notification as usize, Release);
+    fn store_release(&self, notification: Notification, strategy: Option<NotifyOneStrategy>) {
+        let data: usize = match strategy {
+            None => notification as usize & NOTIFICATION_TYPE_MASK,
+            Some(strategy) => {
+                (((strategy as usize) << NOTIFICATION_NOTIFY_ONE_STRATEGY_SHIFT)
+                    & NOTIFICATION_NOTIFY_ONE_STRATEGY_MASK)
+                    | (notification as usize & NOTIFICATION_TYPE_MASK)
+            }
+        };
+        self.0.store(data, Release);
     }
 
-    fn load(&self, ordering: Ordering) -> Option<Notification> {
-        match self.0.load(ordering) {
+    fn load(&self, ordering: Ordering) -> (Option<Notification>, Option<NotifyOneStrategy>) {
```

Comment:
Is there ever a time when this will return `None` for `Notification` but will return `Some` for `NotifyOneStrategy`? I don't think there is. Perhaps, `Notification` the enum should take the strategy as an argument:

```rust
enum Notification {
    One(Strategy),
    All,
}
```

Reply:
Good point, let me change this towards your suggestion.